### PR TITLE
`model.markConfig()` => `compileMarkConfig` & extract `compileStackProperties`

### PR DESCRIPTION
**Pull request by kanitw**
- extract `compileStackProperties` to stack.ts
- add `compileMarkConfig` and remove `model.markConfig()`
- fix `fieldDef.isDimension|Measure` and use it entirely in `model`

Please merge #926 first!  

(See [Branch Specific Diff](https://github.com/uwdata/vega-lite/compare/kw/scale.type...kw/markConfig))

### CODE REVIEW

File: src/compiler/config.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import {StackProperties} from './stack';
+
+import {X, Y, DETAIL} from '../channel';
+import {isAggregate, has} from '../encoding';
+import {isMeasure} from '../fielddef';
+import {POINT, TICK, CIRCLE, SQUARE} from '../mark';
+import {contains, extend} from '../util';
+
+
+export function compileMarkConfig(spec: Spec, stack: StackProperties) {
```
Comment (domoritz) [inline src/compiler/config.ts]: Add a comment that says that we preprocess the config to change the defaults.

File: src/compiler/marks.ts [hunk 2]
```diff
@@ -12,16 +12,16 @@ export function compileMarks(model: Model): any[] {
   const name = model.spec().name;
   const isFaceted = model.has(ROW) || model.has(COLUMN);
   const dataFrom = {data: model.dataTable()};
-  const sortBy = model.markConfig('sortBy');
+  const sortBy = model.config().mark.sortBy;
```
Comment (domoritz) [inline src/compiler/marks.ts]: Create markConfig const.

File: src/compiler/stack.ts [hunk 3]
```diff
@@ -25,6 +34,51 @@ interface StackTransform {
   output: any;
 }
 
+export function compileStackProperties(spec: Spec) {
```
Comment (domoritz) [inline src/compiler/stack.ts]: Add comment

**Comment by kanitw** [comment]
Done.

**Pull request merged**